supervisor/backups/manager.py

```diff
@@ -366,19 +387,24 @@ async def _do_restore(
             return False
         else:
             if addon_start_tasks:
+                change_stage(RestoreJobStage.AWAIT_ADDON_RESTARTS)
                 # Ignore exceptions from waiting for addon startup, addon errors handled elsewhere
                 await asyncio.gather(*addon_start_tasks, return_exceptions=True)
 
             return True
         finally:
-            # Do we need start Home Assistant Core?
-            if not await self.sys_homeassistant.core.is_running():
-                await self.sys_homeassistant.core.start()
+            change_stage(RestoreJobStage.CHECK_HOME_ASSISTANT)
+
+            # Leave Home Assistant alone if it wasn't part of the restore
+            if homeassistant:
+                # Do we need start Home Assistant Core?
+                if not await self.sys_homeassistant.core.is_running():
+                    await self.sys_homeassistant.core.start()
 
-            # Check If we can access to API / otherwise restart
-            if not await self.sys_homeassistant.api.check_api_state():
-                _LOGGER.warning("Need restart HomeAssistant for API")
-                await self.sys_homeassistant.core.restart()
+                # Check If we can access to API / otherwise restart
+                if not await self.sys_homeassistant.api.check_api_state():
+                    _LOGGER.warning("Need restart HomeAssistant for API")
+                    await self.sys_homeassistant.core.restart()
```

Comment:
This code was causing the "Run Supervisor" test to fail. The step to restore the SSH addon only from the backup was getting stuck trying to start Home Assistant. Which makes sense since HA was never supposed to be running in the supervisor CI.

Wrapping it like this fixed it. It also makes much more sense, there's no reason to do anything with HA core when it wasn't even part of the restore.

In general I also question the existence of this code. When you restore HA from a backup the code for updating HA already waits for it to come back online and does a rollback if it doesn't. So what use cases does this bit at the end handle? And is there a particular reason its in the finally and not here (or part of the `task_hass` task):
https://github.com/home-assistant/supervisor/blob/b1380d46ecd99a088ffd74f906359c518b803fcf/supervisor/backups/manager.py#L380-L382

Reply:
It also causes an edge case bug. If someone has stopped HA core and restores a backup from the cli it will start HA core again. If a restore changes the version of HA we have logic to prevent us from starting HA when it was stopped by the user prior to the update here:
https://github.com/home-assistant/supervisor/blob/b1380d46ecd99a088ffd74f906359c518b803fcf/supervisor/homeassistant/core.py#L244-L246

But that code won't actually matter during a restore because we just end up starting HA later anyway due to the code above.

This is definitely a niche edge case. But still it is a small bug. And since I'm not sure what purpose the code does serve I'd like to remove it possible.